core/data_migration_scripts/4_10_0_stages/EE_DMS_4_10_0_Event_Question_Group.dmsstage.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ *
+ * EE_DMS_4_9_0_Email_System_Question
+ *
+ * @package         Event Espresso
+ * @subpackage
+ * @author              Brent Christensen
+ *
+ */
+class EE_DMS_4_10_0_Event_Question_Group extends EE_Data_Migration_Script_Stage_Table
+{
+
+
+
+    /**
+     * Just initializes the status of the migration
+     *
+     * @return EE_DMS_4_9_0_Email_System_Question
+     */
+    public function __construct()
+    {
+        global $wpdb;
+        $this->_pretty_name = __('Event-Question Group Relations', 'event_espresso');
+        $this->_old_table = $wpdb->prefix.'esp_event_question_group';
+        $this->_extra_where_sql = "WHERE EQG_primary = 0";
+        parent::__construct();
+    }
+
+
+
+    /**
+     * updates the question with the new question type
+     * @param array $event_question_group an associative array where keys are column names and values are their values.
+     * @return null
+     */
+    protected function _migrate_old_row($event_question_group)
+    {
+        if (isset($event_question_group['EVT_ID'], $event_question_group['QSG_ID'])) {
+            global $wpdb;
+            $success = $wpdb->update(
+                $this->_old_table,
+                ['EQG_additional' => true],  // data
+                [
+                    'EVT_ID' => $event_question_group['EVT_ID'],
+                    'QSG_ID' => $event_question_group['QSG_ID']
+                ],  // where
```

Comment:
Somehow I didn't see this comment. It's bang-on.
Although since this was made, the situation has changed a little: we now only migrate rows that have `EQG_primary=0 and EQG_additional=0`. That means we only need to deal with scenarios 2 and 3. (Scenario 1 requires no change because the event-question-group row already has the right value; scenario 4 never did require any change).
So I'm updating the code and adding a test for the other two cases. Before, we were handling situation 2 ok but no situation 3.

Reply:
> we now only migrate rows that have `EQG_primary=0 and EQG_additional=0`.

???

how can a row have `EQG_primary=0` **AND** `EQG_additional=0`

I'm assuming that was a typo and you meant `EQG_primary=0` OR `EQG_additional=0`

...err... or maybe you meant `EQG_primary=1` and `EQG_additional=1`

???

Reply:
It's understandable that you're skeptical about that, there should never be a row with both `EQG_primary` and `EQG_additional` equal to 0... except after the column `EQG_additional` was added, and before we remove it during this migration stage.
Here's the situation: in EE 4.9.x we have an event with a question group for additional attendees. So there's a row with `EQG_primary=0`. 
Then, the site is upgraded to 4.10.x, and the column `EQG_additional` is added, with the default value of `0`. Notice that row now has `EQG_primary=0` and `EQG_additional=0`.
Does that answer how we can get that unexpected value?
Of course, we don't want to leave it that way. It's situation 2 or 3. If situation 2, we need to just update that existing row to have `EQG_additional=1`. If situation 3, we need to delete that row, and update the other event-question-group row to have `EQG_additional=1` also.

Reply:
ahhhh kk... ya gotcha... thnx